### PR TITLE
Fix call to ft_resolve_transfer; extend tests

### PR DESCRIPTION
**Pull request by birchmd**
Fixes a bug where the sender/receiver were swapped when refunding the transfer from a failed `ft_transfer_call`. Expands tests to show the correct behaviour now happens.

### CODE REVIEW

**Review by mrLSD — APPROVED** [review]
LGTM!

File: engine-tests/src/tests/eth_connector.rs [hunk 1]
```diff
@@ -518,14 +528,45 @@ fn test_ft_transfer_call_without_message() {
     );
     res.assert_success();
```
Comment (sept-en) [inline engine-tests/src/tests/eth_connector.rs:529]: Why do we assert success?

File: engine-tests/src/tests/eth_connector.rs [hunk 2]
```diff
@@ -1363,3 +1404,11 @@ fn test_ft_transfer_wrong_u128_json_type() {
         "Expected failure as number type can't be parsed to u128",
     );
 }
+
+/// Bytes for a NEAR smart contract implementing `ft_on_transfer`
+fn ft_receiver_bytes() -> Vec<u8> {
```
Comment (sept-en) [inline engine-tests/src/tests/eth_connector.rs:1409]:
```suggestion
fn dummy_ft_receiver_bytes() -> Vec<u8> {
```

File: engine-tests/src/tests/eth_connector.rs [hunk 3]
```diff
@@ -518,14 +528,45 @@ fn test_ft_transfer_call_without_message() {
...
+        json!({
+            "receiver_id": ft_receiver.account_id(),
+            "amount": transfer_amount.to_string(),
+            "msg": "",
+        })
+        .to_string()
+        .as_bytes(),
+        DEFAULT_GAS,
+        1,
+    );
```
Comment (sept-en) [inline engine-tests/src/tests/eth_connector.rs:557]:
```suggestion
    let dummy_ft_receiver = master_account.deploy(
        &dummy_ft_receiver_bytes(),
        "ft-rec".parse().unwrap(),
        near_sdk_sim::STORAGE_AMOUNT,
    );
    let res = recipient_account.call(
        CONTRACT_ACC.parse().unwrap(),
        "ft_transfer_call",
        json!({
            "receiver_id": dummy_ft_receiver.account_id(),
            "amount": transfer_amount.to_string(),
            "msg": "",
        })
        .to_string()
        .as_bytes(),
        DEFAULT_GAS,
        1,
    );
```

File: engine-tests/src/tests/eth_connector.rs [hunk 4]
```diff
@@ -518,14 +528,45 @@ fn test_ft_transfer_call_without_message() {
...
+    res.assert_success();
+
+    let balance = get_eth_on_near_balance(&master_account, DEPOSITED_RECIPIENT, CONTRACT_ACC);
+    assert_eq!(balance, DEPOSITED_AMOUNT - DEPOSITED_FEE - transfer_amount);
+    let balance = get_eth_on_near_balance(
+        &master_account,
+        ft_receiver.account_id().as_ref(),
```
Comment (sept-en) [inline engine-tests/src/tests/eth_connector.rs:564]:
```suggestion
        dummy_ft_receiver.account_id().as_ref(),
```

File: etc/ft-receiver/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    fn ft_on_transfer(
+        &mut self,
+        sender_id: ValidAccountId,
+        amount: U128,
+        msg: String,
+    ) -> PromiseOrValue<U128> {
+        log!("in {} tokens from @{} ft_on_transfer, msg = {}", amount.0, sender_id.as_ref(), msg);
+        PromiseOrValue::Value(U128::from(0))
+    }
+}
```
Comment (sept-en) [inline etc/ft-receiver/src/lib.rs:22]: Let's call it `DummyFungibleTokenReceiver` to emphasize that it actually does nothing in it

**Review by sept-en — APPROVED** [review]
I added some minor comments to it, but this is a really good catch.
Thanks for noticing and fixing this @birchmd !

File: engine-tests/src/tests/eth_connector.rs (hunk 1 above)
Comment (birchmd) [reply engine-tests/src/tests/eth_connector.rs:529]: This is a quirk of the NEAR system. The transaction succeeds overall even though one promise in it failed.

File: engine-tests/src/tests/eth_connector.rs [hunk 6]
```diff
@@ -460,6 +460,10 @@ fn test_ft_transfer_call_eth() {
 #[test]
 fn test_ft_transfer_call_without_message() {
     let (master_account, contract) = init(CUSTODIAN_ADDRESS);
+    let recipient_account = master_account.create_user(
+        DEPOSITED_RECIPIENT.parse().unwrap(),
+        near_sdk_sim::STORAGE_AMOUNT,
```
[the remark this answers is not in the document]
Comment (birchmd) [reply engine-tests/src/tests/eth_connector.rs]: Fixed in https://github.com/aurora-is-near/aurora-engine/pull/330/commits/c72cf6e59e9fd053705c4d7295dddd9e82468aab

File: etc/ft-receiver/src/lib.rs (hunk 5 above)
Comment (birchmd) [reply etc/ft-receiver/src/lib.rs]: Done in https://github.com/aurora-is-near/aurora-engine/pull/330/commits/a9c6a2908ce489696d9c7a6f7ac8a11572e3153f

File: engine-tests/src/tests/eth_connector.rs (hunk 4 above)
Comment (birchmd) [reply engine-tests/src/tests/eth_connector.rs]: Done in https://github.com/aurora-is-near/aurora-engine/pull/330/commits/a9c6a2908ce489696d9c7a6f7ac8a11572e3153f

File: engine-tests/src/tests/eth_connector.rs (hunk 3 above)
Comment (birchmd) [reply engine-tests/src/tests/eth_connector.rs]: Done in https://github.com/aurora-is-near/aurora-engine/pull/330/commits/a9c6a2908ce489696d9c7a6f7ac8a11572e3153f

File: engine-tests/src/tests/eth_connector.rs (hunk 2 above)
Comment (birchmd) [reply engine-tests/src/tests/eth_connector.rs]: Done in https://github.com/aurora-is-near/aurora-engine/pull/330/commits/a9c6a2908ce489696d9c7a6f7ac8a11572e3153f

**Pull request merged**